save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/modal/ModalBuilder.kt

```diff
@@ -0,0 +1,94 @@
+@file:Suppress("HEADER_MISSING_IN_NON_SINGLE_CLASS_FILE")
+
+package com.saveourtool.save.frontend.components.modal
+
+import com.saveourtool.save.frontend.externals.fontawesome.faTimesCircle
+import com.saveourtool.save.frontend.externals.fontawesome.fontAwesomeIcon
+import com.saveourtool.save.frontend.externals.modal.Styles
+import csstype.ClassName
+import react.ChildrenBuilder
+import react.dom.aria.ariaLabel
+import react.dom.html.ButtonType
+import react.dom.html.ReactHTML.button
+import react.dom.html.ReactHTML.div
+import react.dom.html.ReactHTML.h2
+import react.dom.html.ReactHTML.h5
+import react.dom.html.ReactHTML.span
+
+/**
+ * Universal function to create modals with bootstrap styles inside react modals.
+ *
+ * @param isOpen modal openness indicator - should be in state
+ * @param title title of the modal that will be shown in top-left corner
+ * @param message main text that will be shown in the center of modal
+ * @param modalStyle [Styles] that will be applied to react modal
+ * @param onCloseButtonPressed callback that will be applied to `X` button in the top-right corner
+ * @param buttonBuilder lambda that generates several buttons, must contain either [button] or [buttonBuilder]
```

Comment:
> must contain either [button] or [buttonBuilder]

But it's not checked in any way :) I see the intention, however

Reply:
For now I have no idea how to implement it to be both simple and safe. Could pass array of buttons, but not sure it would work nice. Moreover, I do not like that I should explicitly pass `onCloseButtonPressed`, but again I have no better idea